cmd/cli/plugin/cluster/upgrade.go

```diff
@@ -352,6 +349,15 @@ func getClusterTKRNameFromClusterLabels(clusterLabels map[string]string) (string
 }
 
 func getValidTKRVersionFromClusterForUpgrade(cluster *capiv1.Cluster, tkrName string) (string, error) {
+	// if tkrname is not specified, return the latest available version
```

Comment:
From the issue description, we should avoid upgrade from 1.22 to 1.24. How is this latest available version satisfy this condition?

Reply:
this label is tagged by tkr-status-controller https://github.com/vmware-tanzu/tanzu-framework/blob/6c7f5cc18a6caeed6724e88ebbe90ccd05e35e7f/tkr/controller/tkr-status/clusterstatus/clusterstatus.go#L183 . It is selected from tkrs within minor+1 version.